src/util/NuulsUploader.cpp

```diff
@@ -0,0 +1,216 @@
+#include "NuulsUploader.hpp"
+
+#include "common/Env.hpp"
+#include "common/NetworkRequest.hpp"
+#include "providers/twitch/TwitchMessageBuilder.hpp"
+
+#include <QBuffer>
+#include <QHttpMultiPart>
+#include <QMimeDatabase>
+#include <QMutex>
+
+#define UPLOAD_DELAY 2000
+// Delay between uploads in milliseconds
+
+namespace {
+
+boost::optional<QByteArray> convertToPng(QImage image)
+{
+    QByteArray imageData;
+    QBuffer buf(&imageData);
+    buf.open(QIODevice::WriteOnly);
+    bool success = image.save(&buf, "png");
+    if (success)
+    {
+        return boost::optional<QByteArray>(imageData);
+    }
+    else
+    {
+        return boost::optional<QByteArray>(boost::none);
+    }
+}
+}  // namespace
+
+namespace chatterino {
+// These variables are only used from the main thread.
+auto uploadMutex = QMutex();
+std::queue<RawImageData> uploadQueue;
+
+void uploadImageToNuuls(RawImageData imageData, ChannelPtr channel,
+                        ResizingTextEdit &textEdit)
+{
+    const static char *boundary = "thisistheboudaryasd";
```

Comment:
Might be some constexpr variant instead or at least const ptr to const char (with the static too probably)

Reply:
Is `const static char * const boundary` what you wanted me to do? FeelsDonkMan